prov/util/src/util_shm.c

```diff
@@ -486,6 +518,10 @@ void smr_unmap_from_endpoint(struct smr_region *region, int64_t id)
 
 	peer_peers[peer_id].addr.id = -1;
 	peer_peers[peer_id].name_sent = 0;
+
+#if HAVE_XPMEM
+	xpmem_release(local_peers[peer_id].xpmem.apid);
+#endif
```

Comment:
Please look for ways to remove ifdef's, particularly in .c files.  In this case, we can define xpmem_release() as a no-op macro in a header file if HAVE_XPMEM is not defined.